### ISSUE TITLE
hfst-optimized-lookup doesn't handle mutliple fst's in single archive

### ISSUE THREAD

**Issue by snomos** [hfst/hfst#395]
Given the attached hfstol file containing two fst's, the following command gives the following output:

```
$ echo Kapittel | hfst-optimised-lookup -q -n 1 tools/hyphenators/fstbased/hyphenator-gt-desc.hfstol
Kapittel	Kapittel	+?
```

The expected output (and what I get with standard hfst-lookup) is:

```
$ echo Kapittel | hfst-lookup -q -n 1 tools/hyphenators/fstbased/hyphenator-gt-desc.hfstol
Kapittel	Ka^pit^tel	5000,000000
```

The idea is that the first transducer, which is a full, lexical fst, should return the correct hyphenation points for all known words, whereas if no analysis is found, the second fst, which is a plain rule transducer, should still return likely guesses. This works perfectly as intended when using hfst-lookup (where two fst's by default are unioned together, and one can easily discard the non-best results to be left with only the best output), but not when using hfst-optimised-lookup. In that case the second fst is completely ignored, leading to wrong and unexpected output.

[hyphenator-gt-desc.hfstol.zip](https://github.com/hfst/hfst/files/2314376/hyphenator-gt-desc.hfstol.zip)

**Comment by eaxelson** [comment]
Yes it seems that only the first transducer in an archive is used. Now a warning message is printed (commit d69404d). The problem still needs to be fixed.
